lib/ActivityStream.jsm

```diff
@@ -237,6 +237,10 @@ const PREFS_CONFIG = new Map([
     title: "Track spoc impressions",
     value: "{}",
   }],
+  ["discoverystream.rec.impressions", {
+    title: "Track rec impressions",
+    value: "{}",
```

Comment:
#4743 says it doesn't do batching yet, so we could try to implement batching here although it's not entirely clear how to debounce appropriately.

Alternatively, we could add an attribute here along the lines `updateOnlyMain: true` and additionally check that before broadcasting or just plain action:
https://github.com/mozilla/activity-stream/blob/8f09010f3034beb5f988c804719e2b19760bc83f/lib/PrefsFeed.jsm#L49-L50

Reply:
Longer term we might want to flip the condition to `shouldBroadcastUpdates` only for prefs that content cares about as most prefs we don't care but probably not the right time to audit… @k88hudson ?

Reply:
Quick hack seems to avoid content reducer from triggering:
```diff
diff --git a/lib/PrefsFeed.jsm b/lib/PrefsFeed.jsm
   onPrefChanged(name, value) {
     if (this._prefMap.has(name)) {
-      this.store.dispatch(ac.BroadcastToContent({type: at.PREF_CHANGED, data: {name, value}}));
+      this.store.dispatch(ac[name == "discoverystream.rec.impressions" ? "OnlyToMain" : "BroadcastToContent"]({type: at.PREF_CHANGED, data: {name, value}}));
```

Reply:
Is this above hack more desirable than using `Services.prefs.setStringPref`?

Reply:
That seems fine to me, although possibly more risky for an uplift (i'd probably prefer to see it as a non-66 refactor)

Reply:
If we do end up refactoring after 66, then having a different pref name outside of browser.newtabpage.activity-stream will be problematic.. or at least needs some migration.

Reply:
If we keep the pref under the existing branch and want to avoid the broadcasts, we can switch to Services.prefs AND remove the entries from ActivityStream.jsm